ckanext/datapusher/tests/test_interfaces.py

```diff
@@ -0,0 +1,157 @@
+import json
+import httpretty
+import nose
+import sys
+import datetime
+
+from nose.tools import raises
+
+import pylons
+from pylons import config
+import sqlalchemy.orm as orm
+import paste.fixture
+
+from ckan.tests import helpers, factories
+import ckan.plugins as p
+import ckan.lib.create_test_data as ctd
+import ckan.model as model
+import ckan.tests.legacy as tests
+import ckan.config.middleware as middleware
+
+import ckanext.datapusher.interfaces as interfaces
+import ckanext.datastore.db as db
+from ckanext.datastore.tests.helpers import rebuild_all_dbs, set_url_type
+
+
+
+# avoid hanging tests https://github.com/gabrielfalcao/HTTPretty/issues/34
+if sys.version_info < (2, 7, 0):
+    import socket
+    socket.setdefaulttimeout(1)
+
+class FakeDataPusherPlugin(p.SingletonPlugin):
+    p.implements(p.IConfigurable, inherit=True)
+    p.implements(interfaces.IDataPusher, inherit=True)
+
+    def configure(self, config):
+        self.after_upload_calls = 0
+
+    def can_upload(self, resource_id):
+        return False
+
+    def after_upload(self, resource_id):
+        self.after_upload_calls += 1
+
+
+class TestInterace(tests.WsgiAppCase):
+    sysadmin_user = None
+    normal_user = None
+
+    @classmethod
+    def setup_class(cls):
+        wsgiapp = middleware.make_app(config['global_conf'], **config)
+        cls.app = paste.fixture.TestApp(wsgiapp)
+        if not tests.is_datastore_supported():
+            raise nose.SkipTest("Datastore not supported")
+        p.load('datastore')
+        p.load('datapusher')
+        p.load('test_datapusher_plugin')
+
+        ctd.CreateTestData.create()
```

Comment:
CreateTestData iis frowned upon on the new tests. Better to call factories.Sydadmin/User/Dataset ... on the relevant tests